service/provider/src/main/java/gov/va/vro/service/provider/camel/PrimaryRoutes.java

```diff
@@ -64,9 +68,29 @@ private void configureRouteClaimSubmitForFull() {
         .process(FunctionProcessor.fromFunction(saveToDbService::insertClaim))
         // Use Properties not Headers
         // https://examples.javacodegeeks.com/apache-camel-headers-vs-properties-example/
+        .setProperty("id", simple("${body.recordId}"))
+        .setProperty("diagnosticCode", simple("${body.diagnosticCode}"))
+        .setProperty("veteranIcn", simple("${body.veteranIcn}"))
+        .setProperty("claimSubmissionId", simple("${body.claimSubmissionId}"))
+        .setProperty("idType", simple("${body.idType}"))
         .setProperty("diagnosticCode", simple("${body.diagnosticCode}"))
         .routingSlip(method(SlipClaimSubmitRouter.class, "routeClaimSubmit"))
-        .routingSlip(method(SlipClaimSubmitRouter.class, "routeClaimSubmitFull"));
+        .routingSlip(method(SlipClaimSubmitRouter.class, "routeClaimSubmitFull"))
+        .convertBodyTo(String.class)
+        .process(
+            new Processor() {
+              @Override
+              public void process(Exchange exchange) throws Exception {
+                String response = (String) exchange.getIn().getBody();
+                String icn = (String) exchange.getProperty("veteranIcn");
+                UUID id = (UUID) exchange.getProperty("id");
+                String claimSubmissionId = (String) exchange.getProperty("claimSubmissionId");
+                String idType = (String) exchange.getProperty("idType");
+                String diagnosticCode = (String) exchange.getProperty("diagnosticCode");
```

Comment:
claimSubsmissionId and idType is not really used in insertAssessmentResult. veteranIcn and diagnostic code is available from response. I do not think we need these parameters. id is not descriptive enough. maybe claimRecordId?  I really wish claimSubmissionId would be the thing to search for the claim record but that will probably need some indexing changes in the database.